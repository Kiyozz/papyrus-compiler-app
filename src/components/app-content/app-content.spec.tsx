import { shallow } from 'enzyme'
import React from 'react'
import { Route, Switch, Redirect } from 'react-router-dom'
import AppContent from './app-content'

jest.mock('../../redux/sagas/root.saga', () => {
  return function* () {}
})

describe('<AppContent>', () => {
  it('renders correctly', () => {
    const component = (
      <AppContent />
    )

    expect(component).toBeTruthy()
  })

  it('matches snapshot', () => {
    const component = shallow((
      <AppContent />
    ))

    expect(component).toMatchSnapshot()

    const routes = component.find(Route)

    expect(routes).toHaveLength(3)
    expect(routes.at(0)).toHaveProp('path', '/compilation')
    expect(component.find(Switch)).toExist()
    expect(component.find(Redirect)).toExist()
  })
})
