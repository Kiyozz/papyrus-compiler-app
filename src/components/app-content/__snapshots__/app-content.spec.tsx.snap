// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppContent> matches snapshot 1`] = `
<div
  className="app-content"
>
  <Connect(ConnectedRouterWithContext)
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Switch>
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(AppCompilation)",
            "type": [Function],
          }
        }
        key="/compilation"
        path="/compilation"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(AppGroups)",
            "type": [Function],
          }
        }
        key="/groups"
        path="/groups"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(AppSettings)",
            "type": [Function],
          }
        }
        key="/settings"
        path="/settings"
      />
      <Redirect
        to="/compilation"
      />
    </Switch>
  </Connect(ConnectedRouterWithContext)>
</div>
`;
